hooks/useSetMetadata.ts

```diff
@@ -0,0 +1,63 @@
+import { useState } from "react";
+import { useTokenProvider } from "@/providers/TokenProvider";
+import { Address } from "viem";
+import useSignTransaction from "./useSignTransaction";
+import { getPublicClient } from "@/lib/viem/publicClient";
+import { CHAIN, CHAIN_ID } from "@/lib/consts";
+import { zoraCreator1155ImplABI } from "@zoralabs/protocol-deployments";
+import { useUserProvider } from "@/providers/UserProvider";
+import { uploadJson } from "@/lib/arweave/uploadJson";
+import { fetchTokenMetadata } from "@/lib/protocolSdk/ipfs/token-metadata";
+import getTokenInfo from "@/lib/viem/getTokenInfo";
+
+const useSetMetadata = () => {
```

Comment:
Please refactor this hook / file name to match the onchain function being called
- actual: useSetMetadata
- required: useUpdateTokenURI